### ISSUE TITLE
How to normalize to lowercase filename extensions

### ISSUE THREAD

**Issue by rwilkin** [Studio-42/elFinder#2542]
Sorry for the noob question. I am very new to elFinder.
I am trying to get the extensions of uploaded files to be forced to lowercase. I edited the Normalizer plugin.php and added and option `'ext_lowercase' => true` in the $defaults array.  Then I added 
```
if ($opts['ext_lowercase']) {
	$ext = substr($str,strripos($str, '.') + 1);
	if (function_exists('mb_strtolower')) {
		$ext = mb_strtolower($ext, 'UTF-8');
	} else {
		$ext = strtolower($ext);
	}
	$str = substr($str,0,strripos($str, '.')+1).$ext;
}
```

But I can't figure out how to enable the Normalizer plugin, I thought it was enabled by default.

Can anyone point me in the right direction to get this to work? 

Thanks in advanced,

**Comment by nao-pon** [comment]
@rwilkin Please make your elFinderPluginMyNormalizer class.

**Please use [nightly build](https://github.com/Studio-42/elFinder/archive/2.1.zip) to this code.** ( see b41dbb28e029dfff65e40b3890063ad313c063ae )

- php/plugins/MyNormalizer/plugin.php
```php
class elFinderPluginMyNormalizer extends elFinderPluginNormalizer
{
    protected function normalize($str, $opts) {
        $str = parent::normalize($str, $opts);
        if (!empty($opts['ext_lowercase'])) {
            $ext = substr($str,strripos($str, '.') + 1);
            if (function_exists('mb_strtolower')) {
                $ext = mb_strtolower($ext, 'UTF-8');
            } else {
                $ext = strtolower($ext);
            }
            $str = substr($str,0,strripos($str, '.')+1).$ext;
        }
        return $str;
    }
}
```

And enable your `MyNormalizer` plugin.
```php
$opts = array(
    'bind' => array(
        'upload.pre mkdir.pre mkfile.pre rename.pre archive.pre ls.pre' => array(
            'Plugin.MyNormalizer.cmdPreprocess'
        ),
        'upload.presave' => array(
            'Plugin.MyNormalizer.onUpLoadPreSave'
        )
    ),
    // global configure (optional)
    'plugin' => array(
        'MyNormalizer' => array(
            'enable'    => true,
            'nfc'       => true,
            'nfkc'      => true,
            'umlauts'   => false,
            'lowercase' => false,
            'convmap'   => array()
        )
    ),
    // each volume configure (optional)
    'roots' => array(
        array(
            'driver' => 'LocalFileSystem',
            'path'   => '/path/to/files/',
            'URL'    => 'http://localhost/to/files/'
            'plugin' => array(
                'MyNormalizer' => array(
                    'ext_lowercase' => true
                )
            )
        )
    )
);
```

**Comment by rwilkin** [comment]
That worked beautifully! Thank you very much. 
Just an FYI as well - I had to comment out `debug($cmd)` in Normalizer/plugin.php (Line 124). I assume that was left in by mistake?

Thank you very much for your help.

**Comment by nao-pon** [comment]
> Just an FYI as well - I had to comment out debug($cmd) in Normalizer/plugin.php (Line 124). I assume that was left in by mistake?

Yes, That's my mistake. I fixed it (fee065a18734e4a6f99fce493321018d6042ccd6). Thanks! 👍
